File: hello_ros2_cpp/src/hello_ros2.cpp
#include "rclcpp/rclcpp.hpp"

class HelloRos2 : public rclcpp::Node {
public:
    HelloRos2() : Node("hello_ros2") {
        RCLCPP_INFO(this->get_logger(), "start hello ros2 in cpp !!");
    }

    void run() {
        while(rclcpp::ok()) {
            RCLCPP_INFO(this->get_logger(), "hello ros2 in cpp !!");
            sleep(1);
        }
    }

};

int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);

    auto node = std::make_shared<HelloRos2>();
    
    node->run();
    
    rclcpp::shutdown();

    return 0;
}

File: hello_ros2_py/hello_ros2_py/hello_ros2.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import rclpy
from rclpy.node import Node
import time

class HelloRos2(Node):
    def __init__(self, name):
        super().__init__(name)
        self.get_logger().info("start hello_ros2_py node !!")
        
    def run(self):
        try:
            while rclpy.ok():
                self.get_logger().info("hello ros2 in python !!")
                time.sleep(1)
        except KeyboardInterrupt:
            self.get_logger().info("Node was interrupted, shutting down...")
            
            
def main(args=None):
    rclpy.init(args=args)
    node = HelloRos2("hello_ros2_py")
    
    try:
        node.run()
    except KeyboardInterrupt:
        node.destroy_node()
        rclpy.shutdown()
    
if __name__ == '__main__':
    main()